lib/jekyll/utils/internet.rb

```diff
@@ -0,0 +1,35 @@
+module Jekyll
+  module Utils
+    module Internet
+
+      # Public: Determine whether the present device has a connection to
+      # the Internet. This allows plugin writers which require the outside
+      # world to have a neat fallback mechanism for offline building.
+      #
+      # Example:
+      #   if Internet.connected?
+      #     Typhoeus.get("https://pages.github.com/versions.json")
+      #   else
+      #     Jekyll.logger.warn "Warning:", "Version check has been disabled."
+      #     Jekyll.logger.warn "", "Connect to the Internet to enable it."
+      #     nil
+      #   end
+      #
+      # Returns true if a DNS call can successfully be made, or false if not.
+      module_function
+      def connected?
+        !dns("google.com").nil?
```

Comment:
this might be unnecessary, but is google really the right website to ping here? isn't there some sort of website specifically designed for this purpose?

Reply:
example.com?

Reply:
google.com is more available than example.com.  Google is probably the best place to test as it's the most highly available site on the planet.  Just my 2p, but tbqf it's mostly non-relevant, all things considered, how often does example.com get taken out? Only once I know of.